File: docs/WidgetMill/packages/spec/src/nodes/image.ts
import { z } from "zod";
import { BOX_FIT, IMAGE_SOURCE } from "../enums";
import { BindableString } from "../scalars";
import { BindableDimension } from "../complex/dimension";

/** Image — spec §2.6. */
export const ImageNode = z.object({
  type: z.literal("image"),
  props: z
    .object({
      source: z.enum(IMAGE_SOURCE).optional(),
      src: BindableString.optional(),
      width: BindableDimension.optional(),
      height: BindableDimension.optional(),
      fit: z.enum(BOX_FIT).optional(),
    })
    .default({}),
});

File: docs/WidgetMill/flutter/sdui_flutter/lib/src/builders/image.dart
import 'package:flutter/widgets.dart';
import '../model/sdui_node.dart';
import '../parsing/enums.dart';
import '../parsing/parsers.dart';
import '../renderer.dart';

/// Image — spec §2.6.
Widget buildImage(BuildContext context, SduiNode node, SduiRenderer r) {
  final p = node.props;
  final src = (p['src'] ?? '').toString();
  final width = resolveDimension(context, p['width']);
  final height = resolveDimension(context, p['height']);
  final fit = boxFitFrom(p['fit']);

  if (p['source'] == 'asset') {
    return Image.asset(src, width: width, height: height, fit: fit);
  }
  return Image.network(src, width: width, height: height, fit: fit);
}
